src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -28,6 +28,9 @@ public sealed partial class JsonSerializerOptions
         // Stores the JsonTypeInfo factory, which requires unreferenced code and must be rooted by the reflection-based serializer.
         private static Func<Type, JsonSerializerOptions, JsonTypeInfo>? s_typeInfoCreationFunc;
 
+        // Retrieves JsonConverter instances on-demand during serialization or deserialization.
+        private static Func<Type, JsonSerializerOptions, JsonConverter>? s_converterRetrievalFunc;
```

Comment:
It might make sense to only populate this static with the reflection-based factory, and only set or consult the value when `_serializerContext` is set to null.